src/screens/Main/EditStack/SnippetSelection/SnippetSelection.tsx

```diff
@@ -20,120 +20,115 @@ interface Props {
   navigation: NavigationScreenProp<NavigationState>,
 }
 
+const updateSnippetState = function(
+  item: Snippet, 
+  snippetState: Array<Snippet>, 
+  selectedSnippetCount: number, 
+  setSnippetState: Function, 
+  setSelectedSnippetCount: Function) {
+  const currentQuestionState = snippetState
+  // the question is selected previously, remove it from the list and update the order of other items
+  if(currentQuestionState[item.id-1].isSelected){
+    const temp = []
+    const preOrder = item.orderInList
+    for(let i = 0; i < currentQuestionState.length; i++){
+      // move up question in list
+      if(i != (item.id-1)){
+        if(currentQuestionState[i].orderInList > preOrder){  
+          const newQuestion: Snippet = {
+            id: currentQuestionState[i].id, 
+            isSelected: currentQuestionState[i].isSelected, 
+            orderInList: currentQuestionState[i].orderInList-1,
+            text: currentQuestionState[i].text
+          }
+          temp.push(newQuestion)
+        } else{
+          temp.push(currentQuestionState[i])
+        }
+      } else{
+        item.isSelected = false
+        item.orderInList = 0
+        temp.push(item)
+      }
+      // snippet is newly selected, add to list
+      setSnippetState(temp)
+      setSelectedSnippetCount(selectedSnippetCount-1)
+    }
+  } else{
+    const temp = []
+    for(let i = 0; i < currentQuestionState.length; i++){
+      if(i != (item.id-1)){
+        temp.push(currentQuestionState[i])
+      } else{
+        item.isSelected = true
+        item.orderInList = selectedSnippetCount
+        temp.push(item)
+      }
+      setSnippetState(temp)
+      setSelectedSnippetCount(selectedSnippetCount+1)
+    }
+  }
```

Comment:
mentioned this in my last review, but this could be much simpler. you don't need a temp variable. you just grab the state, update it, then save it back
```js
  const modifiedQuestionState = snippetState

  // the question is selected previously, remove it from the list and update the order of other items
  if(modifiedQuestionState[item.id-1].isSelected){
    const preOrder = item.orderInList
    for(let i = 0; i < modifiedQuestionState.length; i++){
      // move up question in list
      if(i != (item.id-1)){
        if(modifiedQuestionState[i].orderInList > preOrder){  
          modifiedQuestionState[i].orderInList = modifiedQuestionState[i].orderInList-1
        }
      } else{
        modifiedQuestionState[i].isSelected = false
        modifiedQuestionState[i].orderInList = 0
      }
      // snippet is newly selected, add to list
      setSnippetState(modifiedQuestionState)
      setSelectedSnippetCount(selectedSnippetCount-1)
    }
  } 
  // newly selected question
  else{
    for(let i = 0; i < modifiedQuestionState.length; i++){
      if(i == (item.id-1)){
        modifiedQuestionState[i].isSelected = true
        modifiedQuestionState[i].orderInList = selectedSnippetCount
      }
      setSnippetState(modifiedQuestionState)
      setSelectedSnippetCount(selectedSnippetCount+1)
    }
  }
```

Reply:
I see! I was thinking in the wrong way. Thanks for this detailed comment!